tests/foreman/cli/test_import.py

```diff
@@ -0,0 +1,110 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Host Collection CLI"""
+from ddt import ddt
+from fauxfactory import gen_string
+from robottelo.common import ssh
+from robottelo.common.helpers import prepare_import_data
+from robottelo.test import CLITestCase
+from robottelo.cli.import_ import Import
+from robottelo.cli.org import Org
+
+
+@ddt
+def csv_to_dataset(csv):
+    """Converts the csv with header as a first entry to a python dict"""
+    keys = csv[0].split(',')
+    del csv[0]
+    dataset = list()
+    for val in csv:
+        dataset.append(
+            dict(zip(keys, val.split(',')))
+        )
+    return dataset
+
+
+class TestImport(CLITestCase):
+    """Import CLI tests.
+
+    All default tests pass no options to the imprt object
+    In such case methods download a default data set from URL
+    specified in robottelo.properties.
+
+    """
+    def test_import_orgs_default(self):
+        """@test: Import all organizations from the default data set
+        (predefined source).
+
+        @feature: Import Organizations
+
+        @assert: 3 Organizations are created
+
+        """
+        files = prepare_import_data()[1]
+        rc1 = Import.organization(
+            {
+                'csv-file': files['users']
+            }
+        )
+        rc2 = ssh.command('cat {0}'.format(files['users']))
+        # pop the array as ssh.command appends 1 empty item at the end
+        rc2.stdout.pop()
+
+        # now to check whether the orgs from csv appeared in sattelite
+        orgs = set(o['name'] for o in Org.list().stdout)
+        imporgs = set(o['organization'] for o in csv_to_dataset(rc2.stdout))
+
+        self.assertEqual(rc1.return_code, 0)
+        self.assertEqual(
+            rc1.stdout,
+            [
+                u'Summary',
+                u'  Created {0} organizations.'.format(len(imporgs)),
+                u''
+            ]
+        )
+        self.assertEqual(False in [o in orgs for o in imporgs], False)
+
+    def test_import_users_default(self):
+        """@test: Import all 3 users from the our default data set (predefined
+        source).
+
+        @feature: Import Users
+
+        @assert: 3 Users created
+
+        """
+        tmpdir, files = prepare_import_data()
+        pwdfile = tmpdir + '/' + gen_string('alpha', 6)
+
+        rc = Import.user(
+            {
+                'csv-file': files['users'],
+                'new-passwords': pwdfile
+            }
+        )
+        self.assertEqual(rc.return_code, 0)
+        self.assertEqual(rc.stdout, [u'Summary', u'  Created 3 users.', u''])
```

Comment:
This makes sense to me. To the best of my knowledge, the UI tests don't have access to proper return codes in the same way that CLI or API tests do. As a result, this may be the best we can do. Also, we *do* have control over the language the Sat uses. Also:

> You can see the tests relying on a specific locale options all over the place, especially UI tests.

When in Rome…